youtube_dl/extractor/audiomack.py

```diff
@@ -37,33 +38,88 @@ class AudiomackIE(InfoExtractor):
         },
     ]
 
+    @staticmethod
+    def create_song_dictionary(api_response, album_url_tag, track_no=0):
+        # All keys are the same in audiomack api and InfoExtractor format
```

Comment:
I didn't change this.  Yes, the dictionary comprehension could be unraveled.  But I think the one liner followed by the if statements makes clear what is expected, followed by what must be done with the unexpected.  Both are clear and equivalent; mine is just a bit more fun.

I don't know what you mean by the id's.  I understand that an id field must be present in the dictionary - the API response from AM conveniently contains a corresponding field that I copy over.